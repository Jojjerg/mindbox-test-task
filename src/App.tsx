import React, { useEffect } from "react";

import Header from "./components/Header/Header";
import TodosCard from "./components/TodosCard/TodosCard";
import { useTodoContext } from "./context/TodoContext";

const App = () => {
  const {
    getTodos,
    addTodo,
    updateTodo,
    deleteCompletedTodos,
    changeFilter,
    todos,
    filter,
  } = useTodoContext();

  useEffect(() => {
    getTodos();
  }, []);

  return (
    <div className="flex flex-col items-center justify-center gap-6">
      <Header />
      <TodosCard
        todos={todos}
        filter={filter}
        addTodo={addTodo}
        updateTodo={updateTodo}
        changeFilter={changeFilter}
        deleteCompletedTodos={deleteCompletedTodos}
      />
    </div>
  );
};

export default App;
