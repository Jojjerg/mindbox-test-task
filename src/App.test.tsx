import { render, screen } from "@testing-library/react";

import App from "./App";
import React from "react";
import TodoProvider from "./context/TodoContext";
import { act } from "react-test-renderer";
import { createRenderer } from "react-test-renderer/shallow";

const renderer = createRenderer();

const defaultComponent = (
  <TodoProvider>
    <App />
  </TodoProvider>
);

const url = "https://jsonplaceholder.typicode.com/todos/";

describe("<App />", () => {
  it("should render and match the snapshot", () => {
    renderer.render(defaultComponent);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it("header renders correctly", () => {
    const { getByTestId } = render(defaultComponent);
    expect(getByTestId("header")).toBeVisible();
  })

  it("todosCard renders correctly", () => {
    const { getByTestId } = render(defaultComponent);
    expect(getByTestId("todos-card")).toBeVisible();
  })
});
