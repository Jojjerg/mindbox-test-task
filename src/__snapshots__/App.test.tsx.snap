// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<Context.Provider
  value={
    {
      "addTodo": [Function],
      "changeFilter": [Function],
      "deleteCompletedTodos": [Function],
      "filter": "All",
      "getTodos": [Function],
      "todos": [],
      "updateTodo": [Function],
    }
  }
>
  <App />
</Context.Provider>
`;
